crates/evm/core/src/utils.rs

```diff
@@ -99,6 +108,122 @@ pub fn gas_used(spec: SpecId, spent: u64, refunded: u64) -> u64 {
     spent - (refunded).min(spent / refund_quotient)
 }
 
+pub fn get_create2_factory_call_inputs(salt: U256, inputs: CreateInputs) -> CallInputs {
+    let calldata = [&salt.to_be_bytes::<32>()[..], &inputs.init_code[..]].concat();
+    CallInputs {
+        contract: DEFAULT_CREATE2_DEPLOYER,
+        transfer: Transfer {
+            source: inputs.caller,
+            target: DEFAULT_CREATE2_DEPLOYER,
+            value: inputs.value,
+        },
+        input: calldata.into(),
+        gas_limit: inputs.gas_limit,
+        context: CallContext {
+            caller: inputs.caller,
+            address: DEFAULT_CREATE2_DEPLOYER,
+            code_address: DEFAULT_CREATE2_DEPLOYER,
+            apparent_value: inputs.value,
+            scheme: CallScheme::Call,
+        },
+        is_static: false,
+        return_memory_offset: 0..0,
+    }
+}
+
+pub fn create2_handler_register<DB: revm::Database, I: InspectorExt<DB>>(
```

Comment:
this needs some docs
and should mention ins with order this should be installed on the evm